bestfirstsearch.java

```diff
@@ -37,7 +37,7 @@ else if(p1.heuristic_value>p2.heuristic_value)
 class graph
 {
 Scanner sc=new Scanner(System.in);
-@NonNegative int v,e;
+int v,e;
```

Comment:
I expect some errors here, since you try to create a new array on line 51 using `v`, and it is not `@NonNegative`

Reply:
I made the changes.Thanks for the review.